services/wireguard/connection/connection.go

```diff
@@ -48,15 +49,15 @@ type Connection struct {
 func (c *Connection) Start(options connection.ConnectOptions) (err error) {
 	var config wg.ServiceConfig
 	if err := json.Unmarshal(options.SessionConfig, &config); err != nil {
-		return err
+		return errors.Wrap(err, "failed to unmarshal connection config")
```

Comment:
For future reference. Try not to do refactoring in the same PR which solves some bug or implements some feature. Not to clutter reading. Do it in separate PR.

Reply:
I was using with wrapping for my debugging during developing. It was a pity to remove. :)